plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/TransportDeleteByQueryAction.java

```diff
@@ -0,0 +1,350 @@
+
+                    if (logger.isDebugEnabled()) {
+                        logger.trace("scroll request [{}] executed: [{}] document(s) returned", scrollId, docs.length);
+                    }
+
+                    if ((docs.length == 0) || (nextScrollId == null)) {
+                        if (logger.isDebugEnabled()) {
+                            logger.trace("scrolling documents terminated");
+                        }
+                        finishHim(scrollId, false, null);
+                        return;
+                    }
+
+                    if (isTimedOut()) {
+                        if (logger.isDebugEnabled()) {
+                            logger.trace("scrolling documents timed out");
+                        }
+                        finishHim(scrollId, true, null);
+                        return;
+                    }
+
+                    // Delete the scrolled documents using the Bulk API
+                    BulkRequest bulkRequest = new BulkRequest();
+                    for (SearchHit doc : docs) {
+                        DeleteRequest delete = new DeleteRequest(doc.index(), doc.type(), doc.id()).version(doc.version());
+                        SearchHitField routing = doc.field("_routing");
+                        if (routing != null) {
+                            delete.routing((String) routing.value());
+                        }
+                        SearchHitField parent = doc.field("_parent");
+                        if (parent != null) {
+                            delete.parent((String) parent.value());
+                        }
+                        bulkRequest.add(delete);
+                    }
+
+                    final int bulkId = bulks.getAndIncrement();
+                    if (logger.isDebugEnabled()) {
+                        logger.trace("executing bulk request [{}] with [{}] deletions", bulkId, bulkRequest.numberOfActions());
+                    }
+
+                    client.bulk(bulkRequest, new ActionListener<BulkResponse>() {
+                        @Override
+                        public void onResponse(BulkResponse bulkResponse) {
+                            onBulkResponse(bulkId, nextScrollId, bulkResponse);
+                        }
+
+                        @Override
+                        public void onFailure(Throwable e) {
+                            onBulkFailure(bulkId, nextScrollId, docs, e);
+                        }
+                    });
+                }
+
+                @Override
+                public void onFailure(Throwable e) {
+                    logger.error("scroll request [{}] failed, now scrolling document(s) is stopped", e, scrollId);
+                    finishHim(scrollId, isTimedOut(), e);
+                }
+            });
+        }
+
+        void onBulkResponse(int bulkId, String scrollId, BulkResponse bulkResponse) {
+            if (logger.isDebugEnabled()) {
```

Comment:
this applies to the rest of the logging statements here, so won't comment on those as well